islandora-support/src/main/java/ca/islandora/alpaca/support/config/PropertyConfig.java

```diff
@@ -0,0 +1,121 @@
+  @PropertySource(value = PropertyConfig.ALPACA_CONFIG_FILE, ignoreResourceNotFound = true)
+})
+public abstract class PropertyConfig {
+
+  public static final String ALPACA_CONFIG_PROPERTY = "alpaca.config";
+  public static final String ALPACA_HOME_PROPERTY = "alpaca.home";
+  public static final String ALPACA_DEFAULT_HOME = "alpaca-home-directory";
+  public static final String ALPACA_DEFAULT_CONFIG_FILE = "file:${" +
+    ALPACA_HOME_PROPERTY + ":" + ALPACA_DEFAULT_HOME + "}/config/alpaca.properties";
+  public static final String ALPACA_CONFIG_FILE = "file:${" + ALPACA_CONFIG_PROPERTY + "}";
+  // static endpoint name for activemq connection
+  protected static final String JMS_ENDPOINT_NAME = "broker";
+  protected static final String MAX_REDELIVERIES_PROPERTY = "error.maxRedeliveries";
+
+  @Value("${" + MAX_REDELIVERIES_PROPERTY + ":5}")
+  private int maxRedeliveries;
+
+  /**
+   * @return the error.maxRedeliveries amount.
+   */
+  public int getMaxRedeliveries() {
+    return maxRedeliveries;
+  }
+
+  /**
+   * Utility function to append various JMS options like concurrentConsumer variables.
+   * @param queueString
+   *   The original topic/queue string
+   * @param concurrentConsumers
+   *   The number of concurrent consumers. -1 means no setting.
+   * @param maxConcurrentConsumers
+   *   The max number of concurrent consumers. -1 means no setting.
+   * @return
+   *   The modified topic/queue string.
+   */
+  public static String addJmsOptions(final String queueString, final int concurrentConsumers,
+                              final int maxConcurrentConsumers) {
+    final StringBuilder builder = new StringBuilder();
+    if (concurrentConsumers > 0) {
+      builder.append("concurrentConsumers=");
+      builder.append(concurrentConsumers);
+    }
+    if (maxConcurrentConsumers > 0) {
+      if (builder.length() > 0) {
+        builder.append("&");
+      }
+      builder.append("maxConcurrentConsumers=");
+      builder.append(maxConcurrentConsumers);
+    }
+    if (builder.length() > 0) {
+      return queueString + (queueString.contains("?") ? '&' : '?') + builder;
+    }
+    return queueString;
+  }
+
+  /**
+   * Utility to add common endpoint options to HTTP endpoints.
+   * @param httpEndpoint
+   *   The http endpoint string.
+   * @param forceAmpersand
+   *   If you want to use this function with a dynamic endpoint, this is whether to force an ampersand to start.
+   * @return
+   *   The modified http endpoint string.
+   */
+  public String addHttpOptions(final String httpEndpoint, final boolean forceAmpersand) {
+    final StringBuilder builder = new StringBuilder();
+    builder.append(httpEndpoint);
+    // Only append ? or & if there is an endpoint.
+    if (builder.length() > 0) {
+      if (httpEndpoint.contains("?") || forceAmpersand) {
+        builder.append("&");
+      } else {
+        builder.append("?");
```

Comment:
PMD's the appends:

```
3 	/home/adam/repos/Alpaca/islandora-support/src/main/java/ca/islandora/alpaca/support/config/PropertyConfig.java 	101 	Avoid appending characters as strings in StringBuffer.append.
4 	/home/adam/repos/Alpaca/islandora-support/src/main/java/ca/islandora/alpaca/support/config/PropertyConfig.java 	103 	Avoid appending characters as strings in StringBuffer.append.
```

Linking to https://pmd.github.io/pmd-6.21.0/pmd_rules_java_performance.html#appendcharacterwithchar

Reply:
Resolved in https://github.com/Islandora/Alpaca/pull/82/commits/5e5d11e3292a7c6458442aacf47b018476640f8a